### PR TITLE
New language detection

### LINKED ISSUE #8796: Composer error message
After last gut pull I get this error message every time I run the command  `bin/composer.phar install --no-dev`

```
Generating optimized autoload files
Deprecation Notice: Class Text_LanguageDetect_ISO639Test located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetect_ISO639Test.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Text_LanguageDetectTest located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetectTest.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class HTMLPurifier_Language_en_x_test located in ./vendor/ezyang/htmlpurifier/library/HTMLPurifier/Language/classes/en-x-test.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Friendica\Util\PidFile located in ./src/Util/Pidfile.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthException located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthConsumer located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthToken located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_HMAC_SHA1 located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_PLAINTEXT located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_RSA_SHA1 located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthRequest located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthServer located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthDataStore located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthUtil located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class ASNValue located in ./library/ASNValue.class.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class ASN_BASE located in ./library/asn1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
```

### Friendica version you encountered the problem
1. git 2020.06-rc
2. php7.4
3. MariaDB 5.5.5-10.1.44-MariaDB-0+deb9u1

### LINKED ISSUE #9348: Composer dependency "text_languagedetect" out of support
- [x] I have searched open and closed issues for duplicates

### Bug Description

The composer dependency `pear/text_languagedetect` is out of support, see
https://pear.php.net/package/Text_LanguageDetect

Therefore no new bugfixings will be added and in case of composer 2.0, the library will brick (like the composer notices already say):
> Deprecation Notice: Class Text_LanguageDetect_ISO639Test located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetect_ISO639Test.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///home/philipp/PhpstormProjects/friendica/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
> Deprecation Notice: Class Text_LanguageDetectTest located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetectTest.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///home/philipp/PhpstormProjects/friendica/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201

### Steps to Reproduce

Install a new Friendica node an run `php bin/composer.phar install`

Actual Result:

composer notices

Expected Result:

no composer problems and further maintenance of our dependency library.

### Platform Info

Friendica Version: 2020.12-dev

Friendica Source: git

PHP version: 7.3

SQL version:

### PR DESCRIPTION
**Pull request by annando**
We are now using a different language detection module. This not also includes much more languages: https://github.com/patrickschur/language-detection#supported-languages

Also now the composer doesn't complain that much anymore.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -64,7 +63,8 @@
 		"npm-asset/perfect-scrollbar": "0.6.16",
 		"npm-asset/textcomplete": "^0.18.2",
 		"npm-asset/typeahead.js": "^0.11.1",
-		"phpseclib/phpseclib": "^2.0"
+		"phpseclib/phpseclib": "^2.0",
```
Comment (nupplaphil) [inline composer.json]: Please move the phpseclib and the new language detection dependency over the npm/browser assets so we can differ between frontend and backend dependencies (I forgot it last time tbh ;-) )

File: src/Model/Item.php [hunk 2]
```diff
@@ -2472,11 +2472,14 @@ public static function addShadowPost($itemid)
...
+		if (!in_array($item['gravity'], [GRAVITY_PARENT, GRAVITY_COMMENT])) {
+			return '';
+		}
+
 		$naked_body = BBCode::toPlaintext($item['body'], false);
 
-		$ld = new Text_LanguageDetect();
-		$ld->setNameMode(2);
-		$languages = $ld->detect($naked_body, 3);
+		$ld = new Language;
```
Comment (nupplaphil) [inline src/Model/Item.php]:
```suggestion
		$ld = new Language();
```

**Review by nupplaphil — CHANGES_REQUESTED** [review]

**Comment by nupplaphil** [comment]
Fixes #9348

File: composer.json (hunk 1 above)
Comment (annando) [reply composer.json]: I guess I have to manually do something after altering that file. I'm not sure, what I actually had to do.

File: src/Model/Item.php (hunk 2 above)
Comment (annando) [reply src/Model/Item.php]: Done

File: composer.json (hunk 1 above)
Comment (nupplaphil) [reply composer.json]: I sorted it :-)

**Review by nupplaphil — APPROVED** [review]

**Comment by annando** [comment]
Now the system says:
```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
```

**Comment by AlfredSK** [comment]
All composer deprecation notices fixed! 🎉

**Comment by MrPetovan** [comment]
I guess this closes #8796 then

**Comment by AlfredSK** [comment]
Fatal error!

```
[03-Oct-2020 21:21:53 Europe/Berlin] PHP Fatal error:  Uncaught Error: Class 'Text_LanguageDetect_ISO639' not found in /var/www/html/libranet/addon/langfilter/langfilter.php:156
Stack trace:
#0 /var/www/html/libranet/src/Core/Hook.php(217): langfilter_prepare_body_content_filter()
#1 /var/www/html/libranet/src/Core/Hook.php(193): Friendica\Core\Hook::callSingle()
#2 /var/www/html/libranet/src/Model/Item.php(3553): Friendica\Core\Hook::callAll()
#3 /var/www/html/libranet/src/Object/Post.php(359): Friendica\Model\Item::prepareBody()
#4 /var/www/html/libranet/src/Object/Post.php(514): Friendica\Object\Post->getTemplateData()
#5 /var/www/html/libranet/src/Object/Post.php(514): Friendica\Object\Post->getTemplateData()
#6 /var/www/html/libranet/src/Object/Post.php(514): Friendica\Object\Post->getTemplateData()
#7 /var/www/html/libranet/src/Object/Post.php(514): Friendica\Object\Post->getTemplateData()
#8 /var/www/html/libranet/src/Object/Thread.php(207): Friendica\Object\Post->getTemplateData()
#9 /var/www/html/libranet/include/conversation.php(685): Friendica\Object\T in /var/www/html/libranet/addon/langfilter/langfilter.php on line 156
```

**Pull request merged**